### DISCUSSION TITLE
Issues about add_volume function.

### DISCUSSION THREAD

**Discussion by yiming989** [pyvista/pyvista#3810] [Q&A] [answered]
We tried to visualize a dataset consisting of the probability density of the hydrogen atom orbital 3d (.vtk file can be downloaded from [here](https://www.it.uu.se/edu/course/homepage/vetvis/ht10/ComputerExercises/data/a1/hydrogen.vtk)). More details about the task can be found [here](https://www.it.uu.se/edu/course/homepage/vetvis/ht10/ComputerExercises/assignment1.shtml).

The rendering window is blank when I used the add_volume() function. The visualization result is correct when I just visualize a slice or a contour of the dataset. Only the volume visualization shows up nothing. Following is my code.

```
filename = "hydrogen.vtk"
data = pv.read(filename)  

pl = pv.Plotter(notebook=False)
pl.set_background("#333333")

my_opacity = [0, 0.2]


actor = pl.add_volume(data, cmap="coolwarm", opacity=my_opacity)


pl.show(cpos="xy")
```

**Comment by banesullivan** [comment]
Would you please also share a report:

```py
import pyvistas as pv
print(pv.Report())
```

**Comment by yiming989** [comment]
Thanks for your quick reply!

Here is the report.

```
--------------------------------------------------------------------------------
  Date: Tue Jan 10 13:55:52 2023 MST

                OS : Darwin
            CPU(s) : 10
           Machine : x86_64
      Architecture : 64bit
               RAM : 16.0 GiB
       Environment : Python
       File system : apfs
        GPU Vendor : Apple
      GPU Renderer : Apple M1 Pro
       GPU Version : 4.1 Metal - 83

  Python 3.10.8 (main, Nov 24 2022, 08:09:04) [Clang 14.0.6 ]

           pyvista : 0.37.0
               vtk : 9.2.2
             numpy : 1.24.1
           imageio : 2.23.0
            scooby : 0.7.0
             pooch : v1.6.0
        matplotlib : 3.6.2
           IPython : 8.8.0
        ipyvtklink : 0.2.3
        jupyterlab : 3.5.2
         pythreejs : Version unknown
--------------------------------------------------------------------------------
```

I'm not sure whether the dataset has a problem. Since I can use the same code to visualize other datasets with UniformGrid data type.

**Comment by banesullivan** [comment, marked as answer]
I tested this locally without issue on my M2 Mac (so similar hardware):

<img width="624" alt="Screen Shot 2023-01-10 at 1 59 27 PM" src="https://user-images.githubusercontent.com/22067021/211661463-18197bbb-1e34-4cb4-a9f8-62befd0aa8cd.png">

Would you please try with a few different `mapper` choices to `add_volume()`? ref https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.add_volume.html#add-volume

Options include: `'fixed_point'`, `'gpu'`, `'open_gl'`, and `'smart'`

For example

```py
pl = pv.Plotter(notebook=False)
pl.set_background("#333333")

my_opacity = [0, 0.2]
actor = pl.add_volume(data, cmap="coolwarm", opacity=my_opacity, mapper='fixed_point')

pl.show(cpos="xy")
```

**Reply by yiming989** [reply]
It's so weird. I tried all options for the `mapper`, but none of them works on my machine.😬

**Reply by akaszynski** [reply]
We just fixed this issue in https://github.com/pyvista/pyvista/pull/3794 and @banesullivan must be using the latest release.

@yiming989, please install using the latest bleeding edge with:

Install with `pip`:
```
pip install https://github.com/pyvista/pyvista/archive/main.zip
```

Otherwise, install with `git` and `pip`
```
git clone https://github.com/pyvista/pyvista.git
cd pyvista
python -m pip install -e .
```

**Reply by yiming989** [reply]
Thank you all @akaszynski @banesullivan very much! I've installed the latest version and my problem is fixed!!

**Reply by akaszynski** [reply]
Excellent. We'll have the fix out in a release by the end of the month.

**Reply by banesullivan** [reply]
Ah my bad! I didn't even realize  #3794 was fixing something so critical... 🤦🏻‍♂️

**Reply by yiming989** [reply]
Thank you guys!!👍
